include/picongpu/fields/incidentField/profiles/DispersivePulse.hpp

```diff
@@ -79,6 +79,63 @@ namespace picongpu
                         // unit: UNIT_TIME^3
                         static constexpr float_X TOD
                             = static_cast<float_X>(Params::TOD_SI / UNIT_TIME / UNIT_TIME / UNIT_TIME);
+
+                        struct LaserParams
+                        {
+                            //! cell position in the internal laser coordinate system
+                            float3_X internPos;
+                            //! Distance from the current cell to the focus in laser propagation direction.
+                            float_X cellToFocusDistance;
+                            //! Initial frequency dependent complex phase
+                            float_X alpha;
+                            //! Center of a frequency's spatial distribution
+                            float_X center;
+                            //! frequency for which the E-value is calculated
+                            float_X omega;
+
+                            /**
+                             * @param totalCellIdx cell index in the total domain (including all moving window slides)
+                             * @param Omega frequency for which the E-value is calculated
+                             */
+                            HDINLINE
+                            LaserParams(floatD_X const& totalCellIdx, float_X const Omega) : omega(Omega)
+                            {
+                                internPos = this->getInternalCoordinates(totalCellIdx);
```

Comment:
`this->` refers to the LaserParams class now, which does not have an `getInternalCoordinates` function